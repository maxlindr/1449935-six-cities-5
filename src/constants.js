export const OfferCardType = {
  FAVORITE: `favorite`,
  NEAR_PLACE: `near_place`,
  CITIES: `cities`
};

export const City = {
  PARIS: `Paris`,
  COLOGNE: `Cologne`,
  BRUSSELS: `Brussels`,
  AMSTERDAM: `Amsterdam`,
  HAMBURG: `Hamburg`,
  DUSSELDORF: `Dusseldorf`
};

export const AuthorizationStatus = {
  NOT_AUTHORIZED: `NOT_AUTHORIZED`,
  AUTHORIZED: `AUTHORIZED`,
  PENDING: `PENDING`
};

export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
  FAVORITES: `/favorites`,
  ERROR: `/error`,
  OFFER: `/offer/:id`,
};

export const APIRoute = {
  OFFERS: `/hotels`,
  LOGIN: `/login`,
  FAVORITE: `/favorite`
};

export const HttpCode = {
  UNAUTHORIZED: 401,
  NOT_FOUND: 404,
};

export const ErrorMessage = {
  NOT_FOUND: `Resource not found`,
  GENERAL: `Oops! Something went wrong`,
  UNAUTHORIZED: `Unauthorized`,
};
