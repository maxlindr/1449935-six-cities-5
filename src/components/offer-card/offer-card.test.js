import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import offer from '../../../__mocks__/offer-mock';
import {OfferCardType} from '../../constants';
import OfferCard from './offer-card';

const renderer = new ShallowRenderer();

describe(`OfferCard должен корректно отображаться`, () => {
  it(`Тип OfferCardType.CITIES`, () => {
    renderer.render(
        <OfferCard
          type={OfferCardType.CITIES}
          offer={offer}
          onMouseOver={() => {}}
          onMouseLeave={() => {}}
        />
    );

    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it(`Тип OfferCardType.NEAR_PLACE`, () => {
    renderer.render(
        <OfferCard
          type={OfferCardType.NEAR_PLACE}
          offer={offer}
          onMouseOver={() => {}}
          onMouseLeave={() => {}}
        />
    );

    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});

