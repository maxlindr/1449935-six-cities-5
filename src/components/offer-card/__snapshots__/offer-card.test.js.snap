// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferCard должен корректно отображаться Тип OfferCardType.CITIES 1`] = `
<article
  className="cities__place-card place-card"
  onMouseLeave={[Function]}
>
  <OfferCardImage
    cardType="cities"
    offer={
      Object {
        "bedrooms": 3,
        "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
        "favorite": false,
        "features": Array [
          "Laptop friendly workspace",
          "Washer",
          "Towels",
          "Fridge",
          "Air conditioning",
          "Baby seat",
          "Breakfast",
        ],
        "host": Object {
          "avatar": "img/avatar-angelina.jpg",
          "id": 25,
          "name": "Angelina",
          "super": true,
        },
        "id": "1",
        "location": Object {
          "city": Object {
            "coordinates": Array [
              50.938361,
              6.959974,
            ],
            "name": "Cologne",
            "zoom": 13,
          },
          "coordinates": Array [
            50.957361,
            6.9509739999999995,
          ],
          "zoom": 16,
        },
        "maxAdults": 4,
        "photos": Array [
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
        ],
        "premium": false,
        "price": 493,
        "rating": 3.8,
        "thumbnail": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
        "title": "The house among olive ",
        "type": "house",
      }
    }
  />
  <OfferCardInfo
    favorites={false}
    offer={
      Object {
        "bedrooms": 3,
        "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
        "favorite": false,
        "features": Array [
          "Laptop friendly workspace",
          "Washer",
          "Towels",
          "Fridge",
          "Air conditioning",
          "Baby seat",
          "Breakfast",
        ],
        "host": Object {
          "avatar": "img/avatar-angelina.jpg",
          "id": 25,
          "name": "Angelina",
          "super": true,
        },
        "id": "1",
        "location": Object {
          "city": Object {
            "coordinates": Array [
              50.938361,
              6.959974,
            ],
            "name": "Cologne",
            "zoom": 13,
          },
          "coordinates": Array [
            50.957361,
            6.9509739999999995,
          ],
          "zoom": 16,
        },
        "maxAdults": 4,
        "photos": Array [
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
        ],
        "premium": false,
        "price": 493,
        "rating": 3.8,
        "thumbnail": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
        "title": "The house among olive ",
        "type": "house",
      }
    }
    roundRating={true}
  />
</article>
`;

exports[`OfferCard должен корректно отображаться Тип OfferCardType.FAVORITE 1`] = `
<article
  className="favorites__card place-card"
  onMouseLeave={[Function]}
>
  <OfferCardImage
    cardType="favorite"
    offer={
      Object {
        "bedrooms": 3,
        "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
        "favorite": false,
        "features": Array [
          "Laptop friendly workspace",
          "Washer",
          "Towels",
          "Fridge",
          "Air conditioning",
          "Baby seat",
          "Breakfast",
        ],
        "host": Object {
          "avatar": "img/avatar-angelina.jpg",
          "id": 25,
          "name": "Angelina",
          "super": true,
        },
        "id": "1",
        "location": Object {
          "city": Object {
            "coordinates": Array [
              50.938361,
              6.959974,
            ],
            "name": "Cologne",
            "zoom": 13,
          },
          "coordinates": Array [
            50.957361,
            6.9509739999999995,
          ],
          "zoom": 16,
        },
        "maxAdults": 4,
        "photos": Array [
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
        ],
        "premium": false,
        "price": 493,
        "rating": 3.8,
        "thumbnail": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
        "title": "The house among olive ",
        "type": "house",
      }
    }
  />
  <OfferCardInfo
    favorites={true}
    offer={
      Object {
        "bedrooms": 3,
        "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
        "favorite": false,
        "features": Array [
          "Laptop friendly workspace",
          "Washer",
          "Towels",
          "Fridge",
          "Air conditioning",
          "Baby seat",
          "Breakfast",
        ],
        "host": Object {
          "avatar": "img/avatar-angelina.jpg",
          "id": 25,
          "name": "Angelina",
          "super": true,
        },
        "id": "1",
        "location": Object {
          "city": Object {
            "coordinates": Array [
              50.938361,
              6.959974,
            ],
            "name": "Cologne",
            "zoom": 13,
          },
          "coordinates": Array [
            50.957361,
            6.9509739999999995,
          ],
          "zoom": 16,
        },
        "maxAdults": 4,
        "photos": Array [
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
        ],
        "premium": false,
        "price": 493,
        "rating": 3.8,
        "thumbnail": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
        "title": "The house among olive ",
        "type": "house",
      }
    }
    roundRating={false}
  />
</article>
`;

exports[`OfferCard должен корректно отображаться Тип OfferCardType.NEAR_PLACE 1`] = `
<article
  className="near-places__card place-card"
  onMouseLeave={[Function]}
>
  <OfferCardImage
    cardType="near_place"
    offer={
      Object {
        "bedrooms": 3,
        "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
        "favorite": false,
        "features": Array [
          "Laptop friendly workspace",
          "Washer",
          "Towels",
          "Fridge",
          "Air conditioning",
          "Baby seat",
          "Breakfast",
        ],
        "host": Object {
          "avatar": "img/avatar-angelina.jpg",
          "id": 25,
          "name": "Angelina",
          "super": true,
        },
        "id": "1",
        "location": Object {
          "city": Object {
            "coordinates": Array [
              50.938361,
              6.959974,
            ],
            "name": "Cologne",
            "zoom": 13,
          },
          "coordinates": Array [
            50.957361,
            6.9509739999999995,
          ],
          "zoom": 16,
        },
        "maxAdults": 4,
        "photos": Array [
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
        ],
        "premium": false,
        "price": 493,
        "rating": 3.8,
        "thumbnail": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
        "title": "The house among olive ",
        "type": "house",
      }
    }
  />
  <OfferCardInfo
    favorites={false}
    offer={
      Object {
        "bedrooms": 3,
        "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
        "favorite": false,
        "features": Array [
          "Laptop friendly workspace",
          "Washer",
          "Towels",
          "Fridge",
          "Air conditioning",
          "Baby seat",
          "Breakfast",
        ],
        "host": Object {
          "avatar": "img/avatar-angelina.jpg",
          "id": 25,
          "name": "Angelina",
          "super": true,
        },
        "id": "1",
        "location": Object {
          "city": Object {
            "coordinates": Array [
              50.938361,
              6.959974,
            ],
            "name": "Cologne",
            "zoom": 13,
          },
          "coordinates": Array [
            50.957361,
            6.9509739999999995,
          ],
          "zoom": 16,
        },
        "maxAdults": 4,
        "photos": Array [
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
          "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
        ],
        "premium": false,
        "price": 493,
        "rating": 3.8,
        "thumbnail": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
        "title": "The house among olive ",
        "type": "house",
      }
    }
    roundRating={false}
  />
</article>
`;
